controller/src/main/java/io/pravega/controller/server/eventProcessor/requesthandlers/DeleteStreamTask.java

```diff
@@ -93,14 +95,14 @@ public DeleteStreamTask(final StreamMetadataTasks streamMetadataTasks,
 
     private CompletableFuture<Void> notifyAndDelete(OperationContext context, String scope, String stream, long requestId) {
         log.info(requestId, "{}/{} deleting segments", scope, stream);
-        return streamMetadataStore.getAllSegmentIds(scope, stream, context, executor)
-                .thenComposeAsync(allSegments -> {
-                    return streamMetadataTasks.notifyDeleteSegments(scope, stream, allSegments, streamMetadataTasks.retrieveDelegationToken(), requestId)
+        return Futures.exceptionallyExpecting(streamMetadataStore.getAllSegmentIds(scope, stream, context, executor)
+                .thenComposeAsync(allSegments -> 
+                    streamMetadataTasks.notifyDeleteSegments(scope, stream, allSegments, streamMetadataTasks.retrieveDelegationToken(), requestId)),
+                            e -> Exceptions.unwrap(e) instanceof StoreException.DataNotFoundException, null)
```

Comment:
We are already handling StoreException.DataNoFoundException in line 80. Hence this check is not required right ?

Reply:
No, this is required because we want to delete stream metadata even if getAllSegments threw DataNotFoundException. 

If trying to delete a stream throws DNF then that is treated as an idempotent case which is handled on line 80.

Reply:
thanks @shiveshr